extensions/one-tab-group/src/index.tsx

```diff
@@ -0,0 +1,264 @@
+
+  return (
+    <List
+      filtering={false}
+      onSearchTextChange={setSearchText}
+      navigationTitle="Search Tabs/Tab Groups"
+      searchBarPlaceholder="Search Saved Tabs/Tab Groups"
+    >
+      {filteredList &&
+        filteredList.map((item: any) => {
+          const hasGroup = item.children && item.children.length > 0;
+
+          return (
+            <List.Item
+              icon={{
+                source: hasGroup ? Icon.CircleFilled : (item.favIconUrl && item.favIconUrl) || Icon.Link,
+                tintColor: hasGroup ? mapCastColor(item.color) : null,
+              }}
+              title={item.title}
+              actions={
+                hasGroup ? (
+                  <ActionPanel>
+                    <Action
+                      title="Explore Tabs In Tab Group"
+                      onAction={() => push(<TabGroupRender group={item.children} />)}
+                    />
+                  </ActionPanel>
+                ) : (
+                  <TabActions tab={item} />
+                )
+              }
+              key={item.id}
+              accessories={hasGroup ? [{ text: `${item.children.length} Tabs` }] : []}
+            />
+          );
+        })}
+    </List>
+  );
+}
+
+function SessionRender(props: { state: State }) {
+  const [menu, setMenu] = useState<Menu>(Menu.Sessions);
+  const [searchText, setSearchText] = useState("");
+  const [filteredList, filterList] = useState(props.state.sessionList);
+  const [loadingSession, setLoadingSession] = useState(props.state.loading);
+
+  const { push } = useNavigation();
+
+  useEffect(() => {
+    let list = props.state.sessionList.filter((item) => item.title.includes(searchText));
+    list = menu === "favorites" ? list.filter((item) => item.starred) : list;
+    filterList(list);
+  }, [searchText, menu, props.state.sessionList]);
+
+  useEffect(() => {
+    setLoadingSession(props.state.loading);
+  }, [props.state.loading]);
+
+  return (
+    <List
+      filtering={false}
+      onSearchTextChange={setSearchText}
+      navigationTitle="Search Sessions"
+      searchBarPlaceholder="Search Saved Sessions"
+      isLoading={loadingSession}
+      searchBarAccessory={
+        <List.Dropdown tooltip="Select Menu" storeValue onChange={(newValue) => setMenu(newValue as Menu)}>
+          {Object.entries(Menu).map(([name, value]) => (
+            <List.Dropdown.Item key={value} title={name} value={value} />
+          ))}
+        </List.Dropdown>
+      }
+    >
+      {filteredList.map((session) => {
+        const { tabs, tabGroups } = flattenTabs(session.tabTree);
+        const createdtimeAgo = dayjs(Number(session.created_at)).fromNow();
+
+        return (
+          <List.Item
+            icon="icon-512.png"
+            title={session.title}
+            subtitle={`Created at ${createdtimeAgo}`}
+            actions={
+              <ActionPanel>
+                <Action
+                  title="Explore Tabs/Tab Groups"
```

Comment:
Can this be both tabs and groups?

Reply:
Yes, Tabs and Tab Group in the some level

Reply:
I don't have a key, but is there a way from the data to determine this? That would be awesome

Reply:
Sorry, This extension is to access browser tab information, which is sensitive information and cannot be provided to you a license key for testing.

Reply:
But since you have a key you might be able to check if the data has a parameter that tells what type it is to the title could be specific instead?

Reply:
I will explain this with an image

In the chrome browser, we can create multiple tab groups, and the groups can contain multiple tabs.

But they are essentially on the same level, so we can both explore/search `Tabs` or `Tab Groups`

![CleanShot 2023-01-11 at 22 01 11@2x](https://user-images.githubusercontent.com/6118824/211826036-d629ca29-2245-4af1-ba93-e587ae28fb2e.png)